### DISCUSSION TITLE
New release?

### DISCUSSION THREAD

**Discussion by SanderRuusmaa** [xiaolin/react-image-gallery#734] [Ideas]
When will the new release come out regarding the QoL changes by replacing defaultProps? This will eliminate errors in the console while developing.

**Comment by xiaolin** [comment]
Will try to get a release out this week. 😬

**Comment by xiaolin** [comment]
Released in https://github.com/xiaolin/react-image-gallery/releases/tag/v1.2.12

**Reply by topovik** [reply]
Version: "react-image-gallery": "^1.2.12"
Error in console:
app-index.js:32 Warning: Item: Support for defaultProps will be removed from memo components in a future major release. Use JavaScript default parameters instead.

**Reply by SanderRuusmaa** [reply]
This could be due to the fact that the ImageGallery.js component is still using defaultProps. Although ImageGallery.js is a class component and it should be "ok" to use defaultProps there. Maybe the warning does not differentiate between Functional and Class components. The warning specifically says that support for using defaultProps in **Functional** Components will be removed [as it is a bad pattern](https://www.youtube.com/watch?v=fnQusZt49_Y&ab_channel=MattPocock).

Maybe the owner (@xiaolin) of react-image-gallery could shed some light onto this and give his own opinion/viewpoint on this.

Rewriting the ImageGallery.js component as a Functional Component without using defaultProps might be a theoretical fix to this :) Though it would be time consuming as hell and a bit much to ask for.

**Reply by xiaolin** [reply]
Can someone provide steps to reproduce this in a new issue? I'm unable to see the warning in react 18.2.0 in local development.

**Reply by SanderRuusmaa** [reply]
Just tested this.

✅ Creating a new React (18.2.0) App and using the ImageGallery component **DOES NOT** create warnings in the console.

❌ Creating a new Next.js (13.4.6) App and using the ImageGallery component in a React Client Component **CREATES** warnings in the console. The recommended App routing was used. React version in the Next.js app is also 18.2.0

Seems that this issue persists when we use this component in a Next js project currently.
